docs/metadata.md

```diff
@@ -492,6 +503,7 @@ Metadata can be created on a folder by calling
 [`folders.addMetadata(folderID, scope, template, metadata, callback)`](http://opensource.box.com/box-node-sdk/jsdoc/Folders.html#addMetadata)
 with a metadata template and an object of key/value pairs to add as metadata.
 
+<!-- sample post_folders_id_metadata_id_id -->
```

Comment:
We're in the process of adding a convenience method for metadata that tries the POST and falls back to PUT if there's a Conflict error; we would love to steer the vast majority of developers to use this method since it guarantees application of the provided key/value pairs.  How should we handle cases like this, where the SDK provides both a (preferred) convenience method, and a low-level method that maps 1:1 to the endpoint?

Reply:
I think this goes into the variations discussion we talked about a while ago. I think we can resolve that as we ship it. Happy to discuss once this goes live.

Reply:
Just to clarify, we could have samples like this:

Standard sample:
`<!-- sample post_folders_id_metadata_id_id -->`

Variation sample:
`<!-- sample post_folders_id_metadata_id_id with_fallback -->`

We can then show both samples, or only the fallback one in the ref docs.

Reply:
I've tagged the old API calls, for now.